src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<Machine> getAll(Class<? extends Machine> type);
```

Comment:
The method `getAll` should be parameterized to return a list of the specific type of `Machine` subclass. Consider changing the return type to `List<T>` to match the generic type parameter `T` of the interface.